src/relations/openfga.py

```diff
@@ -264,6 +279,46 @@ def _on_check_auth_rule_action(self, event):
         except ApiException:
             event.set_results({"error": "failed to perform ofga operation"})
 
+    @log_event_handler(logger)
+    def _on_list_system_admins_action(self, event):
+        """Handle OpenFGA list system admins action.
+
+        Args:
+            event: The event triggered when the action is performed.
+        """
+        if not _check_openfga_relation(self.charm._state, event):
+            return
+
+        openfga_data = self.charm._state.openfga
+        admin_groups = self.charm.config["auth-admin-groups"].split(",")
+        results = {key: [] for key in admin_groups}
+
+        if admin_groups == [""]:
+            event.set_results(
+                {"result": "command succeeded", "output": "no admin groups set in 'auth-admin-groups' config"}
+            )
+
+        try:
+            for admin_group in admin_groups:
+                body = TupleKey(
+                    object=f"group:{admin_group}",
+                )
+
+                response = asyncio.run(
+                    _perform_ofga_api_call(
+                        event=event, openfga_data=openfga_data, body=body, op_type=OFGAOperationType.READ
+                    )
+                )
+
+                for result in response:
+                    # extract format "user:<email>" into "<email>"
+                    user_email = result.key.user.split("user:")[-1]
```

Comment:
Is there a scenario in which OpenFGA admin groups and Temporal `auth-admin-groups` could get out of sync? For example if OpenFGA metadata database went down. It seems there is an assumption that these are aligned in the code - should there be some validation step? My understanding of OpenFGA is not deep so let me know if this is not a possibility :)

Reply:
The purpose of the `auth-admin-groups` config is to specify the list of groups whose members should have system admin access. When requests are made to Temporal, there is logic implemented as part of the Temporal server to query the OpenFGA store and check for group memberships. If the OpenFGA metadata database is down, that means no requests can be made to the Temporal server because identity cannot be verified.

In terms of sync, the OpenFGA store is really the source of truth for this information. If for example I specify the group `admins` as part of the `auth-admin-groups` config, and there are no relations specifying that someone is a member of group `admins`, it simply means no one has access to the system namespace. The `auth-admin-groups` config is used to enrich the logic of the Temporal server in checking for admin group memberships, not to enrich the information stored by OpenFGA.

Reply:
I'm not too sure I understand what you mean. If there was a loss of data on the OpenFGA side but the `auth-admin-groups` is still set to some value (e.g. `admins`), then that means that when checking for members of group `admins` in OpenFGA, no data will be found and system level access will not be granted.

Reply:
In this case would this event not return: `{"result": "command succeeded", "output": ""})` - I guess I was saying that I expect `"no admin groups set in 'auth-admin-groups' config"` as is with the scenario where `auth-admin-groups` is empty. But perhaps this is an edge case, and it is fine as it is with an empty output.

Reply:
Ah I think I understand now. If `auth-admin-groups` had the value `xyz` set, but there were no relations involving the group `xyz` created in the OpenFGA store (i.e. no members added to such group, the group does not exist in the store), then the output will be:

```
{"result": "command succeeded", "output": "
"xyz": [],
"})
```

Indicating that there are no members of group `xyz` in the store. This is because in the OpenFGA store, you don't actually "create a group", you create a relation. If you have not added a relation between a member and a group, then this group does not exist in the store.